lib/opsgenie_alert.rb

```diff
@@ -0,0 +1,93 @@
+require 'httparty'
+require 'active_support/all'
+require 'active_support/time'
+
+class OpsgenieAlert
+
+  attr_reader :alert
+
+  def initialize(alert)
+    @alert = alert
+  end
+
+  def tags
+    alert["tags"]
+  end
+
+  def id
+    alert["id"]
+  end
+
+  def created_at
+    time = time_with_zone(string: alert["createdAt"])
+  end
+
+  def created_at_date
+    alert["createdAt"].to_s.split("T")[0]
+  end
+
+  def saturday?
+    created_at.saturday?
+  end
+
+  def sunday?
+    created_at.sunday?
+  end
+
+  def inhours?
+    time_range_inhours.include?(created_at.hour)
+  end
+
+  def wakinghours_weekday?
+    time_range_wakinghours_weekday.include?(created_at.hour)
+  end
+
+  def wakinghours_weekend?
+    time_range_wakinghours_weekend.include?(created_at.hour)
+  end
+
+  def sleepinghours?
+    time_range_sleepinghours.include?(created_at.hour)
+  end
+
+  def bank_holiday?
+    bank_holidays.include?(created_at_date)
+  end
+
+  private
+
+  def bank_holidays
+    @bank_holidays ||= begin
+      results = HTTParty.get("https://www.gov.uk/bank-holidays.json",
+        :headers => {
+        "Content-Type" => "application/json"
+      })
+
+      events = results["england-and-wales"]["events"]
+      events.map do |event|
+        event["date"]
+      end
+    end
+  end
+
+  def time_with_zone(string:)
```

Comment:
Is the keyword argument doing anything useful here? Could it be a plain argument instead?